src/jit/codegen.h

```diff
@@ -935,6 +935,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     void genHWIntrinsic_R_R_RM_R(GenTreeHWIntrinsic* node, instruction ins);
     void genHWIntrinsic_R_R_R_RM(
         instruction ins, emitAttr attr, regNumber targetReg, regNumber op1Reg, regNumber op2Reg, GenTree* op3);
+    void genBaseIntrinsic(GenTreeHWIntrinsic* node);
```

Comment:
I think that `Base` is an unfortunate name as there is a `Base` class of intrinsics on Arm64 - or would these just be considered part of that class for Arm64? Either way, I woudl think these need an implementation on Arm64.

Reply:
> or would these just be considered part of that class for Arm64

Yes, these are considered part of the `Base` class of intrinsics (given that they are not behind an ISA flag). IIRC, I believe we determined the existing `Base` class we have for ARM should be renamed (it was something like `Arm64Base`, but we should find the API review notes to be sure). In either case, they all share the same `InstructionSet_Base` and `COMPlus_EnableHWIntrinsic` flag (since they are always available on a given supported platform).

> Either way, I woudl think these need an implementation on Arm64.

For all platforms, the `AsType` methods are supported for both XARCH and ARM64 with this PR. For `get_Zero`, however; I only added support for `xarch` right now; as I am not as familiar with the `arm64` architecture.

Reply:
Also noting that, given these are helper intrinsics (rather than actual HWIntrinsics), they will work on ARM64, just using the software fallback until the rest of the instruction support can be added.

Reply:
OK, thanks - that seems OK for now.